File: src/nomanssave/dl.java
package nomanssave;

class dl extends G {
   // $FF: synthetic field
   final dj hl;

   dl(dj var1) {
      this.hl = var1;
   }

   protected String g(String var1) {
      gv var2 = (gv)dj.j(this.hl).getSelectedItem();
      if (var2 == null) {
         return "";
      } else {
         var1 = var1.trim();
         if (!var1.equals(var2.getName())) {
            var2.setName(var1);
            dj.b(this.hl).setText(var1);
         }

         return var1;
      }
   }
}

File: NMSSaveEditor/nomanssave/mixed/dL.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Windows.Forms;
using System.Globalization;

namespace NMSSaveEditor
{

public class dL : DefaultListCellRenderer {
   // $FF: synthetic field
   public dJ hG;

   public dL(dJ var1) {
      this.hG = var1;
   }

   public Component getListCellRendererComponent(ListBox var1, Object var2, int var3, bool var4, bool var5) {
      Label var6 = (Label)base.getListCellRendererComponent(var1, var2, var3, var4, var5);
      if (var2 == null) {
         var6.SetText("");
      } else {
         eM var7 = (eM)var2;
         if (var7.aW()) {
            if (var4) {
               var6.SetBackground(UIManager.getColor("Settlement.positivePerkHighlight"));
            } else {
               var6.SetForeground(UIManager.getColor("Settlement.positivePerkColor"));
            }
         } else if (var4) {
            var6.SetBackground(UIManager.getColor("Settlement.negativePerkHighlight"));
         } else {
            var6.SetForeground(UIManager.getColor("Settlement.negativePerkColor"));
         }
      }
       return var6;
   }
}

}
